lib/src/http/auto_retry.dart

```diff
@@ -170,18 +171,60 @@ class RetryBackOff {
   /// The back-off method to use. One of none, fixed, or exponential.
   final RetryBackOffMethod method;
 
+  /// Whether to enable jitter or not.
+  final bool enableJitter;
+
+  /// The maximum duration between retries.
+  final maxDurationInMs;
+
+  /// The default maximum duration between retries. (5 minutes)
+  static const defaultMaxDurationInMs = 5 * 60 * 1000;
+
+  int _baseBackOffFunction(int numAttempts) {
+    return this.duration.inMilliseconds * pow(2, numAttempts);
+  }
+
+  Duration calculate(int numAttempts) {
+    Random random = new Random();
+    if (enableJitter != null && enableJitter == true) {
+      if (maxDurationInMs != null) {
+        return new Duration(
+            milliseconds: random.nextInt(
+                min(this.maxDurationInMs, _baseBackOffFunction(numAttempts))));
+      } else {
+        return new Duration(
+            milliseconds: random.nextInt(_baseBackOffFunction(numAttempts)));
+      }
+    } else {
+      if (maxDurationInMs != null) {
+        return new Duration(
+            milliseconds:
+                min(this.maxDurationInMs, _baseBackOffFunction(numAttempts)));
+      } else {
+        return new Duration(milliseconds: _baseBackOffFunction(numAttempts));
+      }
+    }
+  }
+
   /// Construct a new exponential back-off representation where [duration] is
   /// the base duration from which each delay will be calculated.
-  const RetryBackOff.exponential(this.duration)
-      : method = RetryBackOffMethod.exponential;
+  const RetryBackOff.exponential(this.duration,
+      {bool this.enableJitter, maxDurationInMs})
```

Comment:
> any unit other than millisecond or second is likely to be impractical

The fact that there are two options says to me that allowing the use of `Duration` is better than restricting it to MS.

> I don't think we should always use it because we don't always want to give consumers the option of using different time units

It was probably not correct to say we should always use `Duration`, but if there is a limited range of acceptable values, you can enforce that while still giving the consumers the convenience of using `Duration`s.

> it introduces additional overhead where simple integer would suffice.

The `Duration` class is relatively simple and supports a `const` constructor - if the consumer uses a `const` duration, it can be optimized at compile time.